plasma_framework/contracts/src/exits/payment/controllers/PaymentStartInFlightExit.sol

```diff
@@ -212,18 +213,18 @@ library PaymentStartInFlightExit {
     function verifyInputTransactionsIncludedInPlasma(StartExitData memory exitData) private view {
         for (uint i = 0; i < exitData.inputTxs.length; i++) {
             (bytes32 root, ) = exitData.controller.framework.blocks(exitData.inputUtxosPos[i].blockNum());
-            bytes32 leaf = keccak256(exitData.inputTxsRaw[i]);
+            bytes32 leaf = keccak256(exitData.inputTxs[i]);
             require(
-                Merkle.checkMembership(leaf, exitData.inputUtxosPos[i].txIndex(), root, exitData.inputTxsInclusionProofs[i]),
-                "Input transaction is not included in plasma"
-            );
+                    Merkle.checkMembership(leaf, exitData.inputUtxosPos[i].txIndex(), root, exitData.inputTxsInclusionProofs[i]),
+                    "Input transaction is not included in plasma"
+                );
         }
     }
 
-    function verifyInputsSpendingCondition(StartExitData memory exitData) private view {
+    function verifyInputsSpent(StartExitData memory exitData) private view {
         for (uint i = 0; i < exitData.inputTxs.length; i++) {
             uint16 outputIndex = exitData.inputUtxosPos[i].outputIndex();
-            bytes32 outputGuard = exitData.inputTxs[i].outputs[outputIndex].outputGuard;
+            WireTransaction.Output memory output = WireTransaction.getOutput(exitData.inputTxs[i], outputIndex);
```

Comment:
Here a "cheat" is done.
Objective of the PR is to abstract from input transaction types. 
This line assumes, that we can extract some "output" data from transaction knowing only that it conforms to wire transaction format (as in Wire Transaction Format Proposal doc).
It would be very handy and make life easier, if all our transaction conforms to that format.
If we don't want to assume that. We could use an interface here, like we do with state transition verifier.

Reply:
hmm, with #214 the new interface probably wouldn't need to getOutput here. However, we do need at least some way to get the `WithdrawData` in `setInputs(...)`. 

For DEX related txs as input tx this assumption seems fine as all tx outputs would be using same format anyway. For `ERC721` I think it *can* work because we are only getting output here, but I doubt the tx format would be as described in the document. I feel when we introduce atomic swap tx for fungible and non fungible token, it would be more or less like the following:

```
{
    txType: int.
    inputs: [bytes32, ...],
    fungibleTokenOutputs: [{ }, { },...],
    nonfungibleTokenOutputs: [{ }, { }, ...],
    // other fields
}
```

Reason of two output lists is that they need different data structure. I don't think current {amount, outputGuard, token} could fit non-fungible token output. And, at least in solidity, you cannot have one outputs array with multiple struct within it.

So, yes, it *can* work as long as we put the fungible token outputs in the second indexed position. However, whether it is a good idea to do would be more controversial. I think it strongly depends on what `WireTransaction` means and what assumption we are talking about when we are introducing such tx format.

Reply:
Let's wait for clarification. We'll update that if needed. As of now I would leave it like it is, as it conforms to Wire Transaction Format doc.

Reply:
sure!

Reply:
Let's just take in WireTransaction as it is for this PR. I am just noting down more concerns with current approach.

During the call with @paulperegud and @pdobacz yesterday, we are talking about that even the `output` for a DEX settlement might not follow this original structure, where we have to make `outputType` explicit on the output itself. So sth like `(outputType, outputGuard/owner, amount, token)`.

TL;DR we might need to have 2 kinds of output in DEX where one send all funds back to user and another keep fund in exchange due to the attack scenerio described [here](https://github.com/omisego/docs/blob/0773484b2a16872e8c730a4f8648d1dcb317a9b1/docs/restricted_custody_mvp1_spec.md#alice-wants-to-withdraw-an-initial-deposit-from-an-offline-venue-but-settlement-occurs). To be able to differentiate which mechanism to parse the `outputGuard/owner` as ch-ch would need to check when collecting txs, we need a flag thus we need the `outputType` being explicit.